combat-trainer.lic

```diff
@@ -2513,6 +2517,10 @@ class GameState
   end
 
   def dance
+    if @dynamic_dance_skill 
+      filtered_skills = weapon_training.reject { |skill, _| $non_dance_skills.include?(skill) }
+      @dance_skill = filtered_skills.min_by { |skill, _| DRSkill.getrank(skill) }.first
```

Comment:
Should this `min_by` exp and rank?

Reply:
Similar to https://github.com/skyleronken/dr-scripts/blob/907c309f5f10ac6aafb9cfa3a2f08890be4c29f5/combat-trainer.lic#L141 is what I was thinking of

Reply:
I removed the exp portion to avoid switching weapons too frequently. You could theoretically enter swapping hell if the mindstate were part of this determination.

Reply:
Ah, good point. Thank you for the explanation.